hack/dind-cluster.sh

```diff
@@ -263,12 +267,161 @@ function build-images() {
 function build-image() {
   local build_root=$1
   local image_name=$2
+  local dockerfile=${3:-}
 
+  if [[ -n "${dockerfile}" ]]; then
+    dockerfile="-f ${dockerfile}"
+  fi
   pushd "${build_root}" > /dev/null
-    ${DOCKER_CMD} build -t "${image_name}" .
+    ${DOCKER_CMD} build -t "${image_name}" ${dockerfile} .
```

Comment:
If `${dockerfile}` originally had a space in it before the `-f` was added this will break.